File: backend_net/ChronosDtnNetApi/Models/SyncLog.cs
using System;

namespace ChronosDtnNetApi.Models;

// Por que: Define a entidade de log de auditoria operacional para janelas de sincronismo e retransmissões.
public class SyncLog
{
    // Por que: Identificador único primário do log de sincronismo.
    public long Id { get; set; }

    // Por que: Referência estrangeira do link orbital de rádio correspondente.
    public long LinkId { get; set; }

    // Por que: Instância de navegação do link mapeado.
    public Link? Link { get; set; }

    // Por que: Instante exato de conclusão do ciclo de sincronização.
    public DateTime SyncTime { get; set; }

    // Por que: Quantidade de bundles que trafegaram com sucesso.
    public int BundlesSent { get; set; }

    // Por que: Quantidade de falhas físicas de rádio simuladas.
    public int Errors { get; set; }

    // Por que: Status do ciclo de sincronização (SUCCESS, PARTIAL, FAILED).
    public string Status { get; set; } = string.Empty;
}

File: backend_java/src/main/java/br/com/fiap/chronosdtn/model/SyncLog.java
package br.com.fiap.chronosdtn.model;

// Por que: Importa as anotações do Jakarta Persistence para persistência ORM de auditoria de rádio.
import jakarta.persistence.*;
// Por que: Importa as anotações do Lombok para reduzir código boilerplate de get/set.
import lombok.*;
// Por que: Permite registrar carimbos de data/hora exatos em que a rotina de auditoria encerrou.
import java.time.LocalDateTime;

// Por que: Define a classe como entidade JPA.
@Entity
// Por que: Associa à tabela de logs de auditoria e reconciliação.
@Table(name = "TB_CHRONOS_SYNC_LOG")
// Por que: Lombok - Getters para logs.
@Getter
// Por que: Lombok - Setters para logs.
@Setter
// Por que: Lombok - Construtor contendo todos os atributos.
@AllArgsConstructor
// Por que: Lombok - Construtor vazio.
@NoArgsConstructor
public class SyncLog {

    // Por que: Código identificador único do log de reconciliação de sinal.
    @Id
    @Column(name = "ID_LOG")
    private Long id;

    // Por que: Canal de rádio (link) utilizado para transmitir o lote de dados.
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_LINK", nullable = false)
    private Link link;

    // Por que: Data/Hora de finalização do processo de auditoria de rádio.
    @Column(name = "DT_SYNC", nullable = false)
    private LocalDateTime syncTime;

    // Por que: Quantidade de pacotes que completaram a transmissão com sucesso.
    @Column(name = "QT_BUNDLES_SENT", nullable = false)
    private Integer bundlesSent;

    // Por que: Quantidade de erros físicos detectados durante o enlace (handshake errors).
    @Column(name = "QT_ERRORS", nullable = false)
    private Integer errorsCount;

    // Por que: Status final da sincronização (ex: SUCCESS, PARTIAL, FAILED).
    @Column(name = "ST_SYNC", nullable = false, length = 20)
    private String syncStatus;
}
